gst/nnstreamer/tensor_transform/tensor_transform.c

```diff
@@ -1390,6 +1425,58 @@ gst_tensor_transform_stand (GstTensorTransform * filter,
   return GST_FLOW_OK;
 }
 
+/**
+ * @brief subrouting for tensor-tranform, "clamp" case.
+ *        : pixel = if (pixel > max) ? max :
+ *                  if (pixel < min) ? min : pixel
+ * @param[in/out] filter "this" pointer
+ * @param[in] inptr input tensor
+ * @param[out] outptr output tensor
+ * @return Gst flow status
+ */
+static GstFlowReturn
+gst_tensor_transform_clamp (GstTensorTransform * filter,
+    const uint8_t * inptr, uint8_t * outptr)
+{
+  tensor_type in_tensor_type = filter->in_config.info.type;
+  tensor_type out_tensor_type = filter->out_config.info.type;
+
+  gsize in_element_size, out_element_size;
+  gulong i, num, data_idx;
+
+  /* let's utilize transform typecast */
+  tensor_transform_operand_s value;
+  gdouble tmp;
+
+  in_element_size = gst_tensor_get_element_size (in_tensor_type);
+  out_element_size = gst_tensor_get_element_size (out_tensor_type);
+  num = gst_tensor_get_element_count (filter->in_config.info.dimension);
+
+  for (i = 0; i < num; ++i) {
+    /* extract the value */
+    data_idx = in_element_size * i;
+    gst_tensor_transform_set_value (filter, &value, in_tensor_type,
+        (gpointer) (inptr + data_idx));
+
+    /* in_tensor_type to double */
+    gst_tensor_transform_typecast_value (filter, &value, _NNS_FLOAT64);
+    gst_tensor_transform_get_value (filter, &value, (gpointer) (&tmp));
+
+    tmp = CLAMP (tmp, filter->data_clamp.min, filter->data_clamp.max);
```

Comment:
This method performs the operation in double even for values of int or float which can become quite slow.
Any reason to with double?

Reply:
Yes, it makes some inefficient, But as this macro does not consider data types, we might have possible side-effects to handle such various data types. This can be fixed in later PR.

Reply:
Yes, as clamp as a macro doesnt need data type, all that's need to be done is to set the type of min/max/tmp to be same.
Later PR also works.